### PR TITLE
PublishSampleMetadata updates

### PR DESCRIPTION
**Pull request by divyamadala30**
Briefly describe changes proposed in this pull request:
- MetadbSample now has field `datasource`
- `datasource` is populated in sampleDataFactory
- PubblishedSampleMetadata now returns `datasource` and `Patient` with `PatientAlias`

### Screenshots
Patient added to the returned sample JSON:

Screen Shot 2021-12-20 at 3 45 52 PM

MetaDbSample node has populates field `datasource`:

Screen Shot 2021-12-20 at 2 54 56 PM

## Crossing T's and dotting I's

Please follow these checklists to help prevent any unexpected issues from being introduced by the changes in this pull request. If an item does not apply then indicate so by surrounding the line item with `~~` to strikethrough the text. See [basic writing and formatting syntax](https://docs.github.com/en/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) for more information.

### Web service and data model checklist

Please follow these checks if any changes were made to any classes in the web, service, or persistence layers.
- [x] Have the endpoints been tested to ensure they have not broken from the changes introduced?
- [x] Have screenshots been uploaded to demonstrate changes made to the response body JSON schema and/or swagger page?
- [x] Were updates made to the mocked incoming request data and/or mocked published request data?
  - [x] [cmo-metadb test data](https://github.com/mskcc/cmo-metadb/tree/master/service/src/test/resources/data)
  - [x] [cmo-metadb-common test data](https://github.com/mskcc/cmo-metadb-common/tree/master/src/test/resources/data)
- ~[ ] Have unit tests been updated in relation to updates to the mocked test data?~

### Neo4j models and database schema checklist:
- [x] Were any of the neo4j persistence models changed?
- [x] Does the graph database produce the expected changes to models, relationships, and/or property names? [provide screenshot of updated elements in graph db]

### Message handlers checklist:
- ~[ ] Were changes introduced in this PR that may affect the workflow of incoming messages?~
- ~[ ] Are messages following the expected workflow when published to the topic(s) changed in this pull request?~
- ~[ ] Were unit tests added to ensure messages are handled as expected?~

If no unit tests were updated or added, then please explain why: Current units tests cover the added fields

Please describe how the workflow and messaging was tested/simulated:

**Your environment**

- NATS [**local**, local docker, dev server, production]
- Neo4j [**local**, local docker, dev server, production]
- MetaDB [**local**, local docker, dev server, production]
- Message publishing simulation [nats cli, docker nats cli, **metadb publisher tool**, other (describe below)]

### General checklist:
- [x] Have tests or has a separate issue that describes the types of tests that should be created been linked or created? If no test is included it should explicitly be mentioned in the PR why there is no test.
- [x] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)

Signed-off-by: Divya Madala 71040191+divyamadala30@users.noreply.github.com

### CODE REVIEW

File: service/src/main/java/org/mskcc/cmo/metadb/service/util/SampleDataFactory.java [hunk 1]
```diff
@@ -32,6 +32,7 @@ public static MetadbSample buildNewResearchSampleFromMetadata(String requestId,
         sample.addSampleAlias(new SampleAlias(sampleMetadata.getPrimaryId(), "igoId"));
         sample.addSampleAlias(new SampleAlias(
                 sampleMetadata.getInvestigatorSampleId(), "investigatorId"));
+        sample.setDatasource("research");
```
Comment (ao508) [inline service/src/main/java/org/mskcc/cmo/metadb/service/util/SampleDataFactory.java:35]: this should be `igo` for datasource, other than that the pr looks good!

**Review by ao508 — CHANGES_REQUESTED** [review]

File: model/src/main/java/org/mskcc/cmo/metadb/model/web/PublishedMetadbSample.java [hunk 2]
```diff
@@ -326,6 +339,14 @@ public void setSampleAliases(List<SampleAlias> sampleAliases) {
         return sampleAliases;
     }
 
+    public MetadbPatient getPatient() {
```
Comment (ao508) [inline model/src/main/java/org/mskcc/cmo/metadb/model/web/PublishedMetadbSample.java:342]: actually I realize another thing - `patientAliases` only should be added to the published sample

**Review by ao508 — CHANGES_REQUESTED** [review]

File: model/src/main/java/org/mskcc/cmo/metadb/model/web/PublishedMetadbSample.java (hunk 2 above)
Comment (ao508) [reply model/src/main/java/org/mskcc/cmo/metadb/model/web/PublishedMetadbSample.java:342]: another thing regarding `patientAliases` - we should add `@JsonIgnore` to the Patient Metadata // where is that coming from?

Comment (ao508) [reply model/src/main/java/org/mskcc/cmo/metadb/model/web/PublishedMetadbSample.java:342]: this is a screen shot of what the mocked published data looks like / what we should have the web service return for the `sampleAliases` and `patientAliases` contents

image

**Comment by divyamadala30** [comment]
New model for PatientAlias in requestJson:

Screen Shot 2021-12-20 at 3 45 52 PM

**Review by ao508 — APPROVED** [review]

**Pull request merged**